AKS + Microk8s - June 2021 release uplift

mdrakiburrahman opened the pull request:
Note that this contains the same June release changes to the Microk8s section as well (which will be functional once the NodePort DC images are rolled out soon).

Also, the enhancements [we discussed in the Microk8s scenario](https://github.com/microsoft/azure_arc/pull/620) has been ported over to this AKS PR.

likamrat commented on azure_arc_data_jumpstart/aks/arm_template/artifacts/DeployPostgreSQL.ps1:
```diff
@@ -42,26 +45,25 @@ Do {
 
-(Get-Content -Path $settingsTemplate) -replace 'arc_postgres',$pgsqlstring | Set-Content -Path $settingsTemplate
+# Retrieving PostgreSQL connection endpoint
+$pgsqlstring = kubectl get postgresql  jumpstartps -n arc -o=jsonpath='{.status.primaryEndpoint}'
```
```suggestion
$pgsqlstring = kubectl get postgresql jumpstartps -n arc -o=jsonpath='{.status.primaryEndpoint}'
```

mdrakiburrahman commented:
@likamrat  PR has been updated to include Microk8s scenario as well (tested with new DC image)

likamrat commented on azure_arc_data_jumpstart/aks/arm_template/aks.json:
```diff
@@ -1,152 +1,150 @@
+      }
+    },
+    "kubernetesVersion": {
+      "defaultValue": "1.18.17",
```
```suggestion
      "defaultValue": "1.19.11",
```

likamrat commented on azure_arc_data_jumpstart/aks/arm_template/artifacts/azuredeploy.parameters.example.json:
```diff
@@ -1,42 +1,45 @@
+      "value": "https://raw.githubusercontent.com/microsoft/azure_arc/main/azure_arc_data_jumpstart/aks/arm_template/"
+    },
+    "kubernetesVersion": {
+      "value": "1.18.17"
```
```suggestion
      "value": "1.19.11"
```

mdrakiburrahman replied:
@likamrat I will test the deployment with the new AKS version once more.

likamrat commented on azure_arc_data_jumpstart/aks/arm_template/azuredeploy.parameters.json:
```diff
@@ -1,42 +1,45 @@
+      "value": "https://raw.githubusercontent.com/microsoft/azure_arc/main/azure_arc_data_jumpstart/aks/arm_template/"
+    },
+    "kubernetesVersion": {
+      "value": "1.18.17"
```
```suggestion
      "value": "1.19.11"
```

likamrat commented on azure_arc_data_jumpstart/aks/arm_template/artifacts/Bootstrap.ps1:
```diff
@@ -62,10 +64,23 @@ Disable-ieESC
 workflow ClientTools_01
         {
-            $chocolateyAppList = 'azure-cli,az.powershell,kubernetes-cli,vcredist140,microsoft-edge,azcopy10,vscode,putty.install,kubernetes-helm'
+            $chocolateyAppList = 'azure-cli,az.powershell,kubernetes-cli,vcredist140,microsoft-edge,azcopy10,vscode,putty.install,kubernetes-helm,grep'
```
grep?

mdrakiburrahman replied:
@likamrat - I find `grep` is useful to have in the client VM when searching for podnames etc. 

e.g. `kubectl get pods --all-namespaces | grep sql` would filter out all pods with `sql` in it, syntax is easier to use than PowerShell native `Select-String`.

It's not essential though, so happy to get it removed.

likamrat replied:
Was just wondering, let's keep it

likamrat commented on azure_arc_data_jumpstart/aks/arm_template/artifacts/DeployPostgreSQL.ps1:
```diff
@@ -1,13 +1,15 @@
 
 # Deployment environment variables
-$controllerName = "Jumpstart-DC"
+$controllerName = "jumpstart-dc"
```
This will also change the readme screenshot. need to update...

mdrakiburrahman replied:
@likamrat no problem. Let me submit a commit with the readme updates - had created a diagram to add onto the AKS scenario.

mdrakiburrahman replied:
Done

likamrat reviewed (changes requested):
- AKS default version 1.18.7 > 1.19.11
- grep?
- minor screenshot update needed following _$controllerName_ value update

mdrakiburrahman commented:
@likamrat thanks. Let me test a deployment with AKS 1.19.11, and take some updated screenshots for the readme.

I will submit a commit soon with the updated readme.

mdrakiburrahman commented:
@likamrat:
* Tested 1.19.11 deployment with DC, MI and PG
* Updated all AKS scenario screenshots with "jumpstart-dc"
* Added 3 diagrams
* Readme text (3 scenarios) updated to include:
  * New VNET.json reference
  * --parameters templateBaseUrl
  * Diagram

likamrat reviewed (approved):
LGTM. Great work!
